packages/shared/src/components/squads/moderation/SquadModerationList.tsx

```diff
@@ -0,0 +1,75 @@
+import React, { ReactElement } from 'react';
+import { Button } from '../../buttons/Button';
+import { ButtonVariant, ButtonSize } from '../../buttons/common';
+import { VIcon } from '../../icons';
+import { useSquadPostModeration } from '../../../hooks/squads/useSquadPostModeration';
+import { useSquadPendingPosts } from '../../../hooks/squads/useSquadPendingPosts';
+import { SquadModerationItem } from './SquadModerationItem';
+import { SourceMemberRole, Squad } from '../../../graphql/sources';
+import { SquadEmptyScreen } from './SquadEmptyScreen';
+
+interface SquadModerationListProps {
+  squad: Squad;
+}
+
+export function SquadModerationList({
+  squad,
+}: SquadModerationListProps): ReactElement {
+  const { onApprove, onReject, isLoading } = useSquadPostModeration();
+  const { data, isFetched } = useSquadPendingPosts(squad?.id);
+
+  if (!data?.length) {
+    if (!isFetched || !squad) {
+      return (
+        <div className="flex flex-col gap-4 p-6">
+          <span className="typo-title3">Loading...</span>
+        </div>
+      );
+    }
+
+    if (squad.currentMember.role === SourceMemberRole.Member) {
+      return (
+        <SquadEmptyScreen
+          Icon={VIcon}
+          title="All done!"
+          description="All caught up! No posts are pending" // check with product what to show here
```

Comment:
Please add ticket/todo here :)

Reply:
Also this is for the member, not the admin?

Reply:
There is also a page for members to see their pending/rejected posts.

Reply:
Just add a TODO here please :)

Reply:
Right. Forgot about this one. Let me create a ticket.